src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityDiode.java

```diff
@@ -112,7 +129,7 @@ protected int getMaxAmperage() {
 
     protected void reinitializeEnergyContainer() {
         long tierVoltage = GTValues.V[getTier()];
-        this.energyContainer = new EnergyContainerHandler(this, tierVoltage * 16, tierVoltage, amps, tierVoltage, amps);
+        this.energyContainer = new EnergyContainerHandler(this, tierVoltage * 16, tierVoltage, isWorkingAllowed ? amps : 0, tierVoltage, isWorkingAllowed ? amps : 0);
```

Comment:
If I'm reading this correctly, the energy inside diode is voided each time the energy container is rebuilt? Is it an intended feature?

Better approach would making custom energy container trait that auto-syncs amperage with the diode MTE, which would prevent redundant re-instantiation of traits.

Reply:
Yes. But thats the case since before this pr and might be the case elsewhere too.
Not sure if this really is a problem since diodes don't hold much and you rarely change the mode.